src/internal_events/splunk_hec.rs

```diff
@@ -44,3 +46,77 @@ impl InternalEvent for SplunkEventEncodeError {
         );
     }
 }
+
+#[cfg(feature = "sources-splunk_hec")]
+mod source {
+    use super::*;
+    use crate::sources::splunk_hec::ApiError;
+
+    #[derive(Debug)]
+    pub struct SplunkHECEventReceived;
```

Comment:
This can be `pub(crate)` as it is internal.

Reply:
No other event struct is `pub(crate)`, so why this one?

Reply:
I think they should all be `pub(crate)` but it hasn't been discussed yet. I'll probably submit a PR for discussion.

I harp on visibility for a few reasons:
1. Bare `pub` prevents the compiler from warning about dead (unused) code.
2. With tighter restrictions, the compiler can make better decisions about things like inlining.
3. With tighter restrictions, the linker can have less work to do by restricting symbol linkage.
4. Tighter restrictions avoid violating encapsulation contracts.

Now, some times we use symbols conditionally based on feature flags, and expressing the right conditionals is difficult and fragile, which makes just using `pub` much easier. Also, since we split our source into a `lib` and seperate `main`, some things have to be made `pub` to cross that boundary.

Reply:
True, I'll change it.

Reply:
Changed.